### DISCUSSION TITLE
Library webexercise: HTML <input> field get's geplaced by "____________________"

### DISCUSSION THREAD

**Discussion by OffByOne37** [quarto-dev/quarto-cli#8412] [Q&A]
### Description

Hello!

I'm currently using a library that itself uses "exams" and "webexercises"-library. 
Those libraries work great when using with output format html, however I need to use "revealjs". 

The problem is, when I want to create a exercise where an input field is required, the library creates this for me:

```
::: {.webex-check .webex-box}

$k$ = <input class='webex-solveme nospaces' data-tol='0.01' size='20' data-answer='["-0.62","-.62"]'/>

$d$ = <input class='webex-solveme nospaces' data-tol='0.01' size='20' data-answer='["-2.5"]'/>

:::
```

This would be correct and it also works if i render my document as a html. However whenever i render it to the format revealjs the <input>-fields get substituted with simply:

`
 = ____________________
`

Does someone know what the problem is?

Thanks!

**Comment by rgaiacs** [comment]
Dear @OffByOne37, can you share a Git repository with a minimal working example of "exams" and "webexercises"-library and your problem?

**Reply by OffByOne37** [reply]
https://github.com/OffByOne37/problem_input_field

**Reply by OffByOne37** [reply]
Not really my example but the same problem happens.

When rendered as html everything works fine, but whenever it get's rendered as revealjs the <input> fields are replaced with underscores.

**Reply by rgaiacs** [reply]
Thanks for the Git repository.

To my knowledge, Quarto internal workflow is `.qmd` -> `.md` -> `.html` where `.qmd` -> `.md` is drive by knitr or Jupyter and `.md` -> `.html` is drive by Pandoc + filters + post processing.

I added

```
execute:
  keep-md: true
```

to the YAML header and run `quarto render example_with_html.qmd` and `quarto render example_with_revealjs.qmd`. `example_with_html.html.md` has

```
Create fill-in-the-blank questions using `fitb()`, providing the answer as the first argument.

- 2 + 2 is <input class='webex-solveme nospaces' size='1' data-answer='["4"]'/>

You can also create these questions dynamically, using variables from your R session.
```

and `example_with_revealjs.revealjs.md` has

```
Create fill-in-the-blank questions using `fitb()`, providing the answer as the first argument.

- 2 + 2 is _

You can also create these questions dynamically, using variables from your R session.
```

Based on this, I will say that the issue is not related with Pandoc.

I believe that quarto is setting some options in knitr that is causing `fitb()` to produce `_` instead of `<input>`.

**Reply by cderv** [reply]
I think it is just that the **webexercices** R package is not supporting revealjs output. 

It will only check for `html` or `pdf` as output format. But when `format: revealjs` is passed, the value will be `revealjs`
the code do an exact comparison: 
https://github.com/PsyTeachR/webexercises/blob/01ea77227fe39b9b4c208517038521c227ff1eb3/R/webexercises_fns.R#L78-L85

and is not using `knitr::is_html_output()` which would cover revealjs format being part of HTML supported format. 

Maybe `webex.js` is not working with revealjs framework as the two JS libs could conflict 🤷 

Anyhow, this needs to be ask at https://github.com/PsyTeachR/webexercises

**Reply by OffByOne37** [reply]
Big thanks to both of you!

Will ask it on [GitHub-Webexercises](https://github.com/PsyTeachR/webexercises)

**Comment by mcanouil** [comment]
@OffByOne37 Could you edit the title to make it clear it is about "webexercises"?

**Reply by OffByOne37** [reply]
Done:)

**Reply by mcanouil** [reply]
Thanks!
